zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/ZooKeeperEndpointGroup.java

```diff
@@ -59,66 +62,95 @@ public ZooKeeperEndpointGroup(String zkConnectionStr, String zNodePath, int sess
     /**
      * Create a ZooKeeper-based {@link EndpointGroup}, endpoints will be retrieved from a node's all children's
      * node value using {@link NodeValueCodec}.
+     *
      * @param zkConnectionStr a connection string containing a comma separated list of {@code host:port} pairs,
      *                        each corresponding to a ZooKeeper server
      * @param zNodePath       a zNode path e.g. {@code "/groups/productionGroups"}
-     * @param sessionTimeout  Zookeeper session timeout in milliseconds
-     * @param nodeValueCodec  the nodeValueCodec
+     * @param sessionTimeout  ZooKeeper session timeout in milliseconds
+     * @param nodeValueCodec  the {@link NodeValueCodec}
      */
     public ZooKeeperEndpointGroup(String zkConnectionStr, String zNodePath, int sessionTimeout,
                                   NodeValueCodec nodeValueCodec) {
-        zooKeeperConnector = new ZooKeeperConnector(zkConnectionStr, zNodePath, sessionTimeout,
-                                                    createListener());
+        requireNonNull(zNodePath, "zNodePath");
```

Comment:
No need to check `zkConnectionStr`?

Reply:
If there are any problems, `CuratorFrameworkFactory` raises the error.

Reply:
Got it.!